corehq/apps/app_manager/views/releases.py

```diff
@@ -222,32 +231,42 @@ def delete_copy(request, domain, app_id):
 def odk_install(request, domain, app_id, with_media=False):
     app = get_app(domain, app_id)
     qr_code_view = "odk_qr_code" if not with_media else "odk_media_qr_code"
+    build_profile_id = request.GET.get('profile')
+    profile_url = app.odk_profile_display_url if not with_media else app.odk_media_profile_display_url
+    if build_profile_id:
+        profile_url += '?profile={profile}'.format(profile=build_profile_id)
     context = {
         "domain": domain,
         "app": app,
-        "qr_code": reverse("corehq.apps.app_manager.views.%s" % qr_code_view, args=[domain, app_id]),
-        "profile_url": app.odk_profile_display_url if not with_media else app.odk_media_profile_display_url,
+        "qr_code": reverse("corehq.apps.app_manager.views.%s" % qr_code_view,
+                           args=[domain, app_id],
+                           params={'profile': build_profile_id}),
+        "profile_url": profile_url,
     }
     return render(request, "app_manager/odk_install.html", context)
 
 
 def odk_qr_code(request, domain, app_id):
-    qr_code = get_app(domain, app_id).get_odk_qr_code()
+    profile = request.GET.get('profile')
+    qr_code = get_app(domain, app_id).get_odk_qr_code(build_profile_id=profile)
     return HttpResponse(qr_code, content_type="image/png")
 
 
 def odk_media_qr_code(request, domain, app_id):
-    qr_code = get_app(domain, app_id).get_odk_qr_code(with_media=True)
+    profile = request.GET.get('profile')
+    qr_code = get_app(domain, app_id).get_odk_qr_code(with_media=True, build_profile_id=profile)
     return HttpResponse(qr_code, content_type="image/png")
 
 
 def short_url(request, domain, app_id):
-    short_url = get_app(domain, app_id).get_short_url()
+    build_profile_id = request.GET.get('profile')
+    short_url = get_app(domain, app_id).get_short_url(build_profile_id=build_profile_id)
```

Comment:
Looks like this will generate a new URL for any arbitrary profile that is requested, regardless of whether it identifies a valid build profile or not. Does that matter? The same applies to other similar view functions in this file.